### PR TITLE
Add support of the new branch setup to the version update automation

### PR DESCRIPTION
**Pull request by rdner**
## What does this PR do?

So, we can support the new 9.0 release, 8.x branch and the rest of the release branches.

## Why is it important?

It's fixing wrong automation behavior, for example like here https://github.com/elastic/elastic-agent/pull/5953 and here https://github.com/elastic/elastic-agent/pull/5948

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
~- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~
~- [ ] I have added an integration test or an E2E test~

### CODE REVIEW

**Comment by rdner** [comment]
Switching to draft, still needs some work.

File: pkg/testing/tools/git/git.go [hunk 1]
```diff
@@ -26,17 +27,45 @@ type outputReader func(io.Reader) error
 // current repository ordered descending by creation date.
 // e.g. 8.13, 8.12, etc.
 func GetReleaseBranches(ctx context.Context) ([]string, error) {
-	c := exec.CommandContext(ctx, "git", "branch", "-r", "--list", "*/[0-9]*.*[0-9x]", "--sort=-creatordate")
+	c := exec.CommandContext(ctx, "git", "for-each-ref", "--sort=-committerdate", "refs/heads/[0-9]*.*[0-9x]", "--format=%(refname:short)")
```
[the remark this answers is not in the document]
Comment (rdner) [reply pkg/testing/tools/git/git.go]: yeah, the PR was supposed to be a draft and I accidentally requested reviews. But now it's ready for review.

File: pkg/testing/tools/git/git.go [hunk 2]
```diff
@@ -26,17 +27,45 @@ type outputReader func(io.Reader) error
...
+		// 8.16 vs 8.x
+		case err1 == nil && err2 != nil:
+			return false
+		// 8.x vs 8.16
+		case err1 != nil && err2 == nil:
+			return true
+		// 8.x vs 9.x
+		case err1 == nil && err2 == nil:
+			n1, _ = strconv.Atoi(strings.TrimSuffix(branches[i], ".x"))
+			n2, _ = strconv.Atoi(strings.TrimSuffix(branches[j], ".x"))
```
[the remark this answers is not in the document]
Comment (rdner) [reply pkg/testing/tools/git/git.go]: This implementation was completely wrong, I've re-worked it in https://github.com/elastic/elastic-agent/pull/5957/commits/74e4d71d1e19e5cc9469674d7f4f0f6cd26e42d5

**Review by swiatekm — APPROVED** [review]

File: pkg/testing/tools/git/git_test.go [hunk 3]
```diff
@@ -24,3 +28,29 @@ func TestGetReleaseBranches(t *testing.T) {
 		assert.Regexp(t, releaseBranchRegexp, b)
 	}
 }
+
+func TestLess(t *testing.T) {
+	branchList := []string{
+		"8.16",
+		"9.1",
+		"8.x",
+		"wrong",
```
Comment (pkoutsovasilis) [inline pkg/testing/tools/git/git_test.go:37]: can you add also `8wrong.wrong8` and `8.wrong8`?

File: pkg/testing/tools/git/git.go [hunk 4]
```diff
@@ -26,17 +30,44 @@ type outputReader func(io.Reader) error
 // current repository ordered descending by creation date.
 // e.g. 8.13, 8.12, etc.
 func GetReleaseBranches(ctx context.Context) ([]string, error) {
-	c := exec.CommandContext(ctx, "git", "branch", "-r", "--list", "*/[0-9]*.*[0-9x]", "--sort=-creatordate")
+	c := exec.CommandContext(ctx, "git", "for-each-ref", "refs/heads/[0-9]*.*[0-9x]", "--format=%(refname:short)")
```
Comment (pkoutsovasilis) [inline pkg/testing/tools/git/git.go]:
```suggestion
	c := exec.CommandContext(ctx, "git", "for-each-ref", "refs/heads/[0-9]*.[0-9x]*", "--format=%(refname:short)")
```
just wondering if the second `*` be after the `[0-9x]`?!

**Review by pkoutsovasilis — APPROVED** [review]
LGTM

**Pull request merged**